### DISCUSSION TITLE
want to know how to extract font glyph?

### DISCUSSION THREAD

**Discussion by CodePythonFollow** [sumatrapdfreader/sumatrapdf#3579] [Q&A]
When I used pymupdf to obtain some cid-like files, I found that some glyph information was missing. I would like to know how the project obtains glyphs similar to text. I would be very grateful if you could give me some ideas.

**Comment by GitHubRulesOK** [comment]
Best to ask Artifex (MuPDF maintainers) basically in a normal file the binary numbers are a straight conversion to the corresponding font entry where the vectors can be read so pseudo wise if a 16bit unicode character 01000101 01001010 will be a binary code token say for A then the font for that area will be used to ask that font for its A metrics to be scaled to screen, or used by a plain text extraction as single character A etc. 
However you seem to be aware its not that simple with PDF where the font can be many different formats such as glyphless (no plain text equivalent for each token) or Unicode or custom mapped or CID tables thus the look-up more prone to challenges by text extractors. 

Basically PDF was designed to draw shapes like a CAD or Graphics application rather than be reversed into keyboard entries.

**Comment by CodePythonFollow** [comment]
Thank you.
